### PR TITLE
Feature/drop before win7 support in compiler

### PR DESCRIPTION
**Pull request by dothebart**
windows 7 is now the oldest supported target os by the compiler, some more warnings fixed.

### CODE REVIEW

**Comment by dothebart** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1348/

**Comment by dothebart** [comment]
`resilience_sharddist` failed to shutdown one dbserver; this should be unrelated to this pr; shell-foxx-manifest-spec is also unrelated.

**Comment by dothebart** [comment]
fix compilation, more casts, http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1357/

File: lib/Basics/StringUtils.cpp [hunk 1]
```diff
@@ -294,11 +294,11 @@ size_t levenshteinDistance(std::vector<uint32_t>& vect1, std::vector<uint32_t>&
...
 
   if (lhsSize < std::numeric_limits<uint8_t>::max()) {
-    return static_cast<size_t>(::levenshtein<uint32_t, uint8_t>(l, r, lhsSize, rhsSize));
+    return static_cast<size_t>(::levenshtein<uint32_t, uint8_t>(l, r, static_cast<uint8_t>(lhsSize), static_cast<uint8_t>(rhsSize)));
   } else if (lhsSize < std::numeric_limits<uint16_t>::max()) {
-    return static_cast<size_t>(::levenshtein<uint32_t, uint16_t>(l, r, lhsSize, rhsSize));
+    return static_cast<size_t>(::levenshtein<uint32_t, uint16_t>(l, r, static_cast<uint16_t>(lhsSize), static_cast<uint16_t>(rhsSize)));
   } else if (lhsSize < std::numeric_limits<uint32_t>::max()) {
-    return static_cast<size_t>(::levenshtein<uint32_t, uint32_t>(l, r, lhsSize, rhsSize));
+    return static_cast<size_t>(::levenshtein<uint32_t, uint32_t>(l, r, static_cast<uint32_t>(lhsSize), static_cast<uint8_t>(rhsSize)));
```
Comment (jsteemann) [inline lib/Basics/StringUtils.cpp]: should be "uint32_t" here.

**Comment by jsteemann** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/1364/

**Pull request merged**